go/vt/schemamanager/schemaswap/schema_swap.go

```diff
@@ -75,6 +91,283 @@ type shardSchemaSwap struct {
 	healthWaitingChannel *chan interface{}
 }
 
+// shardSwapMetadata contains full metadata about schema swaps on a certain shard.
+type shardSwapMetadata struct {
+	err error
+	// IDs of last started and last finished schema swap on the shard
+	lastStartedSwap, lastFinishedSwap uint64
+	// currentSQL is the SQL of the currentlr running schema swap (if there is any)
+	currentSQL string
+}
+
+// Run is the main entry point of the schema swap process. It drives the process from start
+// to finish, including possible restart of already started process. In the latter case the
+// method should be just called again and it will pick up already started process. The only
+// input argument is the SQL statements that comprise the schema change that needs to be
+// pushed using the schema swap process.
+func (schemaSwap *Swap) Run(sql string) error {
+	if err := schemaSwap.createShardObjects(); err != nil {
+		return err
+	}
+	if err := schemaSwap.initializeSwap(sql); err != nil {
+		return err
+	}
+	errHealthWatchers := schemaSwap.runOnAllShards(
+		func(shard *shardSchemaSwap) error {
+			return shard.startHealthWatchers()
+		})
+	// Note: this defer statement is before the error is checked because some shards may
+	// succeed while others fail. We should try to stop health watching on all shards no
+	// matter if there was some failure or not.
+	defer schemaSwap.stopAllHealthWatchers()
+	if errHealthWatchers != nil {
+		return errHealthWatchers
+	}
+	err := schemaSwap.runOnAllShards(
+		func(shard *shardSchemaSwap) error {
+			return shard.applySeedSchemaChange(sql)
+		})
+	if err != nil {
+		return err
+	}
```

Comment:
Your suggestion is getting converted by gofmt into this:
<pre>
	if err := schemaSwap.runOnAllShards(func(shard *shardSchemaSwap) error {
		return shard.applySeedSchemaChange(sql)
	}); err != nil {
		return err
	}
</pre>
This is way beyond readable. I'll keep the code as is.